### PR TITLE
Force users to specify FRDM-K64F IP config

### PR DESCRIPTION
**Pull request by mbolivar**
This series allows users to specify local-only configuration, and recommends that as a mechanism to specify the network configuration on FRDM-K64F. When the network config is unspecified, break the build on that board.

### CODE REVIEW

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Failed.

**Comment by mbolivar** [comment]
The application build fail is intentional. The CI environment needs to specify a boards/frdm_k64f-local.conf with the IP addresses for that environment:

```
/home/jenkins/workspace/Github_PR_Linaro_FOTA_Hawkbit_Application_Build_Checker/PLATFORM/frdm_k64f/zephyr-project-master-upstream-dev/include/toolchain/common.h:107:41: error: size of array '__build_assert_failure' is negative
 #define BUILD_ASSERT(EXPR) typedef char __build_assert_failure[(EXPR) ? 1 : -1]
                                         ^
/home/jenkins/workspace/Github_PR_Linaro_FOTA_Hawkbit_Application_Build_Checker/PLATFORM/frdm_k64f/src/tcp.c:39:1: note: in expansion of macro 'BUILD_ASSERT'
 BUILD_ASSERT(sizeof(CONFIG_NET_SAMPLES_MY_IPV4_ADDR) > 1);
 ^
/home/jenkins/workspace/Github_PR_Linaro_FOTA_Hawkbit_Application_Build_Checker/PLATFORM/frdm_k64f/zephyr-project-master-upstream-dev/include/toolchain/common.h:107:41: error: size of array '__build_assert_failure' is negative
 #define BUILD_ASSERT(EXPR) typedef char __build_assert_failure[(EXPR) ? 1 : -1]
                                         ^
/home/jenkins/workspace/Github_PR_Linaro_FOTA_Hawkbit_Application_Build_Checker/PLATFORM/frdm_k64f/src/tcp.c:41:1: note: in expansion of macro 'BUILD_ASSERT'
 BUILD_ASSERT(sizeof(CONFIG_NET_SAMPLES_PEER_IPV4_ADDR) > 1);
 ^
```

**Comment by mbolivar** [comment]
I think we'd better really solve this problem and not merge half-measures.

I've proposed a static assert macro that emits a message upstream: https://gerrit.zephyrproject.org/r/#/q/topic:build-assert-msg

Will resubmit this on top of that if it gets good feedback.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by mbolivar** [comment]
Upstream static assert patches are merged; this patch series now updated to use them. We'd need to wait for our next Zephyr rebase to merge this still.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Comment by gatek33p3r** [comment]
Application Build Failed.

File: src/tcp.c [hunk 1]
```diff
@@ -23,6 +24,25 @@
...
+ * character, the default Kconfig string value).
+ *
+ * Users can specify these values for their local environment using
+ * local.conf or boards/$(BOARD)-local.conf, which aren't tracked in
+ * Git.
+ */
+#if defined(CONFIG_NET_IPV4)
+#if !defined(CONFIG_NET_DHCPV4)
+BUILD_ASSERT_MSG(sizeof(CONFIG_NET_SAMPLES_MY_IPV4_ADDR) > 1,
+		 "DHCPv4 must be enabled, or CONFIG_NET_SAMPLES_MY_IPV4_ADDR must be defined, in boards/$(BOARD)-local.conf");
```
Comment (mike-scott) [inline src/tcp.c]: Long line here.  Not sure why checkpatch doesn't warn on this.

File: src/tcp.c [hunk 2]
```diff
@@ -23,6 +24,25 @@
...
+#endif
+BUILD_ASSERT_MSG(sizeof(CONFIG_NET_SAMPLES_PEER_IPV4_ADDR) > 1,
+		 "CONFIG_NET_SAMPLES_PEER_IPV4_ADDR must be defined in boards/$(BOARD)-local.conf");
```
Comment (mike-scott) [inline src/tcp.c]: Long line here.  Not sure why checkpatch doesn't warn on this.

File: src/tcp.c (hunk 1 above)
Comment (mbolivar) [inline src/tcp.c]: Done on purpose. Breaking up long string literals is an antipattern IMO. Makes the message harder to grep. Checkpatch might feel the same way.

Comment (mike-scott) [inline src/tcp.c]: Ok, sounds good to me.

**Comment by gatek33p3r** [comment]
Checkpatch Suceeded.

**Comment by gatek33p3r** [comment]
Application Build Failed.

**Comment by gatek33p3r** [comment]
Bootloader Build Succeeded.

**Pull request merged**